# 1. Importing modules

In [49]:
import pandas as pd

# 2. Loading and exploring the data

In this section the goals are to load the data into python and then to check its basic properties. This will include the dimension, column types and names, and missingness counts.

In [50]:
df = pd.read_csv('BKB_WaterQualityData_2020084.csv')
df.head() #head of the data sheet

Site_Id Unit_Id  Read_Date  Salinity (ppt)  Dissolved Oxygen (mg/L)  \
0     Bay     NaN   1/3/1994             1.3                     11.7   
1     Bay     NaN  1/31/1994             1.5                     12.0   
2     Bay     NaN   2/7/1994             1.0                     10.5   
3     Bay     NaN  2/23/1994             1.0                     10.1   
4     Bay     NaN  2/28/1994             1.0                     12.6   

   pH (standard units)  Secchi Depth (m)  Water Depth (m)  Water Temp (?C)  \
0                  7.3              0.40             0.40              5.9   
1                  7.4              0.20             0.35              3.0   
2                  7.2              0.25             0.60              5.9   
3                  7.4              0.35             0.50             10.0   
4                  7.2              0.20             0.40              1.6   

   Air Temp-Celsius  Air Temp (?F) Time (24:00) Field_Tech DateVerified  \
0               8.0          46.40        11:00        NaN          NaN   
1               2.6          36.68        11:30        NaN          NaN   
2               7.6          45.68         9:45        NaN          NaN   
3               2.7          36.86          NaN        NaN          NaN   
4               0.0          32.00        10:30        NaN          NaN   

  WhoVerified  AirTemp (C)  Year  
0         NaN          8.0  1994  
1         NaN          2.6  1994  
2         NaN          7.6  1994  
3         NaN          2.7  1994  
4         NaN          0.0  1994

In [51]:
df.tail() #Tail of the dataset

Site_Id Unit_Id   Read_Date  Salinity (ppt)  Dissolved Oxygen (mg/L)  \
2366     Bay     NaN  10/11/2018             1.9                      5.0   
2367     Bay     NaN  10/24/2018             0.0                      9.0   
2368     Bay     NaN  10/28/2018             0.9                      2.9   
2369     Bay     NaN   11/7/2018             1.7                      NaN   
2370     Bay     NaN  12/11/2018             0.1                      NaN   

      pH (standard units)  Secchi Depth (m)  Water Depth (m)  Water Temp (?C)  \
2366                  7.0              4.00              1.2             25.0   
2367                  7.0              0.30              0.6             18.0   
2368                  7.0              0.40              0.9             13.0   
2369                  7.0              0.45              0.9             20.0   
2370                  7.0              0.10              0.1             10.0   

      Air Temp-Celsius  Air Temp (?F) Time (24:00) Field_Tech DateVerified  \
2366               NaN           78.0        09:30    Sue Poe   11/13/2019   
2367               NaN           58.0        09:30    Sue Poe   11/13/2019   
2368               NaN           49.0        09:20    Sue Poe   11/13/2019   
2369               NaN           65.0        09:45    Sue Poe   11/13/2019   
2370               NaN           42.0        09:40    Sue Poe   11/13/2019   

          WhoVerified  AirTemp (C)  Year  
2366  Christine Folks    25.555556  2018  
2367  Christine Folks    14.444444  2018  
2368  Christine Folks     9.444444  2018  
2369  Christine Folks    18.333333  2018  
2370  Christine Folks     5.555556  2018

In [52]:
df[df.Year.eq(2018)]

Site_Id Unit_Id   Read_Date  Salinity (ppt)  Dissolved Oxygen (mg/L)  \
2208       A     NaN   1/16/2018             0.0                      NaN   
2209       A     NaN   2/14/2018             0.0                      NaN   
2210       A     NaN    3/7/2018             0.0                      NaN   
2211       A     NaN   3/22/2018             0.0                      8.0   
2213       A     NaN   4/17/2018             1.0                      NaN   
...      ...     ...         ...             ...                      ...   
2366     Bay     NaN  10/11/2018             1.9                      5.0   
2367     Bay     NaN  10/24/2018             0.0                      9.0   
2368     Bay     NaN  10/28/2018             0.9                      2.9   
2369     Bay     NaN   11/7/2018             1.7                      NaN   
2370     Bay     NaN  12/11/2018             0.1                      NaN   

      pH (standard units)  Secchi Depth (m)  Water Depth (m)  Water Temp (?C)  \
2208                  7.0              0.60              0.6              5.0   
2209                  6.5              1.00              1.8              7.0   
2210                  6.5              1.10              1.8              8.0   
2211                  6.5              0.70              0.9              7.0   
2213                  7.0              1.30              1.6             14.0   
...                   ...               ...              ...              ...   
2366                  7.0              4.00              1.2             25.0   
2367                  7.0              0.30              0.6             18.0   
2368                  7.0              0.40              0.9             13.0   
2369                  7.0              0.45              0.9             20.0   
2370                  7.0              0.10              0.1             10.0   

      Air Temp-Celsius  Air Temp (?F) Time (24:00) Field_Tech DateVerified  \
2208               NaN           37.0        11:05    Sue Poe   10/29/2019   
2209               NaN           45.0        10:50    Sue Poe   10/29/2019   
2210               NaN           47.0        11:00    Sue Poe   10/29/2019   
2211               NaN           40.0        11:03    Sue Poe   10/29/2019   
2213               NaN           48.0        10:55    Sue Poe   10/29/2019   
...                ...            ...          ...        ...          ...   
2366               NaN           78.0        09:30    Sue Poe   11/13/2019   
2367               NaN           58.0        09:30    Sue Poe   11/13/2019   
2368               NaN           49.0        09:20    Sue Poe   11/13/2019   
2369               NaN           65.0        09:45    Sue Poe   11/13/2019   
2370               NaN           42.0        09:40    Sue Poe   11/13/2019   

          WhoVerified  AirTemp (C)  Year  
2208  Christine Folks     2.777778  2018  
2209  Christine Folks     7.222222  2018  
2210  Christine Folks     8.333333  2018  
2211  Christine Folks     4.444444  2018  
2213  Christine Folks     8.888889  2018  
...               ...          ...   ...  
2366  Christine Folks    25.555556  2018  
2367  Christine Folks    14.444444  2018  
2368  Christine Folks     9.444444  2018  
2369  Christine Folks    18.333333  2018  
2370  Christine Folks     5.555556  2018  

[82 rows x 17 columns]

# 3. Query for years between 1999 & 2018

In [53]:
df1 = df.query('Year >= 1999 & Year <= 2018')

In [54]:
df1.head()

Site_Id Unit_Id   Read_Date  Salinity (ppt)  Dissolved Oxygen (mg/L)  \
345       A     NaN   8/11/1999             3.0                     2.60   
346       A     NaN   8/18/1999             3.0                     1.60   
347       A     NaN    9/8/1999             1.8                     1.35   
348       A     NaN  10/15/1999             0.1                     5.65   
349       A     NaN  11/10/1999             0.0                     6.70   

     pH (standard units)  Secchi Depth (m)  Water Depth (m)  Water Temp (?C)  \
345                  7.7              0.60             0.60             26.0   
346                  8.7               NaN              NaN             31.0   
347                  8.2              0.40             1.00             28.0   
348                  7.3              0.60             0.80             20.0   
349                  7.6              0.65             0.95             18.0   

     Air Temp-Celsius  Air Temp (?F) Time (24:00)    Field_Tech DateVerified  \
345               NaN          80.96        23:02  Not Recorded          NaN   
346               NaN          91.40         9:36  Not Recorded          NaN   
347               NaN          76.60        14:24  Not Recorded          NaN   
348               NaN            NaN         0:00  Not Recorded          NaN   
349               NaN            NaN         0:00  Not Recorded          NaN   

    WhoVerified  AirTemp (C)  Year  
345         NaN    27.200000  1999  
346         NaN    33.000000  1999  
347         NaN    24.777778  1999  
348         NaN   -17.777778  1999  
349         NaN   -17.777778  1999

In [55]:
df1.tail()

Site_Id Unit_Id   Read_Date  Salinity (ppt)  Dissolved Oxygen (mg/L)  \
2366     Bay     NaN  10/11/2018             1.9                      5.0   
2367     Bay     NaN  10/24/2018             0.0                      9.0   
2368     Bay     NaN  10/28/2018             0.9                      2.9   
2369     Bay     NaN   11/7/2018             1.7                      NaN   
2370     Bay     NaN  12/11/2018             0.1                      NaN   

      pH (standard units)  Secchi Depth (m)  Water Depth (m)  Water Temp (?C)  \
2366                  7.0              4.00              1.2             25.0   
2367                  7.0              0.30              0.6             18.0   
2368                  7.0              0.40              0.9             13.0   
2369                  7.0              0.45              0.9             20.0   
2370                  7.0              0.10              0.1             10.0   

      Air Temp-Celsius  Air Temp (?F) Time (24:00) Field_Tech DateVerified  \
2366               NaN           78.0        09:30    Sue Poe   11/13/2019   
2367               NaN           58.0        09:30    Sue Poe   11/13/2019   
2368               NaN           49.0        09:20    Sue Poe   11/13/2019   
2369               NaN           65.0        09:45    Sue Poe   11/13/2019   
2370               NaN           42.0        09:40    Sue Poe   11/13/2019   

          WhoVerified  AirTemp (C)  Year  
2366  Christine Folks    25.555556  2018  
2367  Christine Folks    14.444444  2018  
2368  Christine Folks     9.444444  2018  
2369  Christine Folks    18.333333  2018  
2370  Christine Folks     5.555556  2018

# 4. Converting the slice to CSV file

In [56]:
df1.to_csv(r'C:\Users\ukaja\Documents\Ashwini_Telukuntla_Data_Mining\HW0\back_bay_1998-2018_clean.csv', index = False, header=True)

# 5. How many and what are the names of the columns in this dataset?

In [57]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1956 entries, 345 to 2370
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Site_Id                  1956 non-null   object 
 1   Unit_Id                  32 non-null     object 
 2   Read_Date                1956 non-null   object 
 3   Salinity (ppt)           1867 non-null   float64
 4   Dissolved Oxygen (mg/L)  1150 non-null   float64
 5   pH (standard units)      1872 non-null   float64
 6   Secchi Depth (m)         1891 non-null   float64
 7   Water Depth (m)          1890 non-null   float64
 8   Water Temp (?C)          1842 non-null   float64
 9   Air Temp-Celsius         46 non-null     float64
 10  Air Temp (?F)            1904 non-null   float64
 11  Time (24:00)             1905 non-null   object 
 12  Field_Tech               1956 non-null   object 
 13  DateVerified             386 non-null    object 
 14  WhoVerified           

In [58]:
df1.describe()

Salinity (ppt)  Dissolved Oxygen (mg/L)  pH (standard units)  \
count     1867.000000              1150.000000          1872.000000   
mean         0.410204                 5.867704             7.106490   
std          0.873802                 2.054290             0.799353   
min          0.000000                 0.000000             0.300000   
25%          0.000000                 4.300000             6.500000   
50%          0.000000                 5.900000             7.000000   
75%          0.100000                 7.300000             7.500000   
max          8.000000                11.200000             9.900000   

       Secchi Depth (m)  Water Depth (m)  Water Temp (?C)  Air Temp-Celsius  \
count       1891.000000      1890.000000      1842.000000         46.000000   
mean           0.558958         0.768701        18.096417         14.682609   
std            0.467386         0.615332         8.413927         13.818784   
min            0.000000         0.010000         0.000000          3.000000   
25%            0.300000         0.400000        11.000000          9.000000   
50%            0.500000         0.645000        19.000000         11.000000   
75%            0.700000         1.000000        25.000000         16.000000   
max            7.400000         8.500000        74.000000         74.000000   

       Air Temp (?F)  AirTemp (C)         Year  
count    1904.000000  1956.000000  1956.000000  
mean       61.604748    15.537219  2008.134969  
std        15.668014    10.201950     5.419882  
min        10.500000   -17.777778  1999.000000  
25%        48.000000     8.888889  2004.000000  
50%        63.000000    16.666667  2008.000000  
75%        75.000000    23.888889  2013.000000  
max        91.400000    33.000000  2018.000000

## 6. What is the mean Dissolved Oxygen (mg/L) over the entire dataset?

Mean of Dissolved Oxygen (mg/L) is 5.867704 from df1.describe

## 7. Which year were the highest number of AirTemp (C) data points collected?

From above we came to know that the highest number of AirTemp(C) is 33 from df1.describe

In [59]:
df1.columns = [c.replace(' ', '_') for c in df1.columns]

In [60]:
df1.columns = [c.replace('(', '') for c in df1.columns]

In [61]:
df1.columns = [c.replace(')', '') for c in df1.columns]

In [74]:
df1.columns = [c.replace('/', '') for c in df1.columns]

In [62]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1956 entries, 345 to 2370
Data columns (total 17 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Site_Id                1956 non-null   object 
 1   Unit_Id                32 non-null     object 
 2   Read_Date              1956 non-null   object 
 3   Salinity_ppt           1867 non-null   float64
 4   Dissolved_Oxygen_mg/L  1150 non-null   float64
 5   pH_standard_units      1872 non-null   float64
 6   Secchi_Depth_m         1891 non-null   float64
 7   Water_Depth_m          1890 non-null   float64
 8   Water_Temp_?C          1842 non-null   float64
 9   Air_Temp-Celsius       46 non-null     float64
 10  Air_Temp_?F            1904 non-null   float64
 11  Time_24:00             1905 non-null   object 
 12  Field_Tech             1956 non-null   object 
 13  DateVerified           386 non-null    object 
 14  WhoVerified            386 non-null    object 
 15  Ai

In [64]:
df1.query('AirTemp_C == 33')

Site_Id Unit_Id  Read_Date  Salinity_ppt  Dissolved_Oxygen_mg/L  \
346       A     NaN  8/18/1999           3.0                    1.6   

     pH_standard_units  Secchi_Depth_m  Water_Depth_m  Water_Temp_?C  \
346                8.7             NaN            NaN           31.0   

     Air_Temp-Celsius  Air_Temp_?F Time_24:00    Field_Tech DateVerified  \
346               NaN         91.4       9:36  Not Recorded          NaN   

    WhoVerified  AirTemp_C  Year  
346         NaN       33.0  1999

1999 is the year highest number of AirTemp (C) data points collected

## 8. Which year were the least number of AirTemp (C) data points collected?

From the above we know the least number of AirTemp (C) data points collected is -17.777778 from df1.describe

In [65]:
df1.query('AirTemp_C <= -17')

Site_Id Unit_Id   Read_Date  Salinity_ppt  Dissolved_Oxygen_mg/L  \
348        A     NaN  10/15/1999           0.1                   5.65   
349        A     NaN  11/10/1999           0.0                   6.70   
354        B     NaN    9/8/1999           0.9                   3.60   
355        B     NaN  10/15/1999           1.0                   4.70   
357        B     NaN  11/10/1999           1.0                   6.30   
362        C     NaN    9/8/1999           1.0                   2.40   
364        C     NaN  10/15/1999           0.1                   5.50   
366        C     NaN  11/10/1999           0.2                  10.00   
371        D     NaN    9/8/1999           0.0                   1.50   
372        D     NaN   9/22/1999           0.0                   1.90   
373        D     NaN  10/15/1999           0.0                   3.25   
380      Bay     NaN   1/23/1999           1.1                  11.20   
381      Bay     NaN    8/5/1999           1.8                   1.80   
386      Bay     NaN  10/27/1999           0.3                   4.30   
393        A     NaN    4/3/2000           1.0                    NaN   
415        B     NaN    4/3/2000           1.5                    NaN   
437        C     NaN    4/3/2000           1.0                    NaN   
459        D     NaN    4/3/2000           0.0                   7.40   
481      Bay     NaN    4/3/2000           1.5                   7.60   
558        C     NaN   7/20/2001           NaN                    NaN   
587        D     NaN  10/22/2001           0.0                    NaN   
616        A     NaN   2/11/2002           2.0                    NaN   
635        A     NaN  12/16/2002           0.0                    NaN   
638        B     NaN   2/11/2002           2.0                    NaN   
655        B     NaN  12/16/2002           0.0                    NaN   
658        C     NaN   2/11/2002           2.0                   8.40   
677        C     NaN  12/16/2002           0.0                   5.60   
680        D     NaN   2/11/2002           0.0                   8.30   
699        D     NaN  12/16/2002           0.0                   5.20   
702      Bay     NaN   2/11/2002           3.0                   5.70   
865        B     NaN  11/18/2004           0.0                   5.30   
948        A     NaN    7/1/2005           0.0                    NaN   
949        A     NaN   7/14/2005           0.0                   2.20   
968        B     NaN    7/1/2005           0.0                    NaN   
969        B     NaN   7/14/2005           0.0                   3.50   
989        C     NaN    7/1/2005           0.0                   5.10   
990        C     NaN   7/14/2005           0.0                    NaN   
996        C     NaN  10/27/2005           0.0                   5.00   
1010       D     NaN    7/1/2005           0.0                   2.70   
1011       D     NaN   7/14/2005           0.0                    NaN   
1026     Bay     NaN    3/3/2005           1.0                    NaN   
1033     Bay     NaN    7/1/2005           0.0                   4.30   
1129       D     NaN   7/18/2006           0.0                   2.70   
1140     Bay     NaN    1/3/2006           0.0                   0.80   
1288       A     NaN   6/19/2008           1.0                   5.20   
1312       B     NaN   6/19/2008           1.0                   5.90   
1335       C     NaN   6/19/2008           2.0                    NaN   
1359       D     NaN   6/19/2008           1.0                    NaN   
1538       B     NaN    2/5/2010           NaN                    NaN   
1584       D     NaN   3/16/2010           NaN                    NaN   
1619     Bay     NaN  10/14/2010           0.0                   3.90   
2200       A     NaN  11/29/2017           NaN                    NaN   

      pH_standard_units  Secchi_Depth_m  Water_Depth_m  Water_Temp_?C  \
348                 7.3            0.60           0.80        

1999
2000
2001
2002
2004
2005
2006
2008
2010
2017
Above are the years least number of AirTemp (C) data points collected which is -17.777778

## 9. Which sampling location has the highest mean ppt? What is the equivalent ppm?

In [66]:
df1.query('Salinity_ppt == 0.4')

Site_Id Unit_Id   Read_Date  Salinity_ppt  Dissolved_Oxygen_mg/L  \
374       D     NaN  10/27/1999           0.4                    3.4   

     pH_standard_units  Secchi_Depth_m  Water_Depth_m  Water_Temp_?C  \
374                7.4            0.95           1.35           42.5   

     Air_Temp-Celsius  Air_Temp_?F Time_24:00    Field_Tech DateVerified  \
374               NaN         72.0       0:00  Not Recorded          NaN   

    WhoVerified  AirTemp_C  Year  
374         NaN  22.222222  1999

Based on the df1.describe() mean of ppt is 0.4. Location D has the highest mean ppt and equivalent ppm is 3.4

## 10. When looking at the mean ppt, which location would you infer is furthest from the influence of ocean water inflows? (Assume that higher salinity correlates to closer proximity to the ocean.)

In [67]:
df1.describe()

Salinity_ppt  Dissolved_Oxygen_mg/L  pH_standard_units  Secchi_Depth_m  \
count   1867.000000            1150.000000        1872.000000     1891.000000   
mean       0.410204               5.867704           7.106490        0.558958   
std        0.873802               2.054290           0.799353        0.467386   
min        0.000000               0.000000           0.300000        0.000000   
25%        0.000000               4.300000           6.500000        0.300000   
50%        0.000000               5.900000           7.000000        0.500000   
75%        0.100000               7.300000           7.500000        0.700000   
max        8.000000              11.200000           9.900000        7.400000   

       Water_Depth_m  Water_Temp_?C  Air_Temp-Celsius  Air_Temp_?F  \
count    1890.000000    1842.000000         46.000000  1904.000000   
mean        0.768701      18.096417         14.682609    61.604748   
std         0.615332       8.413927         13.818784    15.668014   
min         0.010000       0.000000          3.000000    10.500000   
25%         0.400000      11.000000          9.000000    48.000000   
50%         0.645000      19.000000         11.000000    63.000000   
75%         1.000000      25.000000         16.000000    75.000000   
max         8.500000      74.000000         74.000000    91.400000   

         AirTemp_C         Year  
count  1956.000000  1956.000000  
mean     15.537219  2008.134969  
std      10.201950     5.419882  
min     -17.777778  1999.000000  
25%       8.888889  2004.000000  
50%      16.666667  2008.000000  
75%      23.888889  2013.000000  
max      33.000000  2018.000000

In [77]:
df1.query('Salinity_ppt == 0')

Site_Id Unit_Id   Read_Date  Salinity_ppt  Dissolved_Oxygen_mgL  \
349        A     NaN  11/10/1999           0.0                  6.70   
365        C     NaN  10/27/1999           0.0                  3.70   
371        D     NaN    9/8/1999           0.0                  1.50   
372        D     NaN   9/22/1999           0.0                  1.90   
373        D     NaN  10/15/1999           0.0                  3.25   
...      ...     ...         ...           ...                   ...   
2362       D     NaN  10/24/2018           0.0                   NaN   
2363       D     NaN  10/28/2018           0.0                   NaN   
2364       D     NaN   11/7/2018           0.0                  6.90   
2365       D     NaN  12/11/2018           0.0                   NaN   
2367     Bay     NaN  10/24/2018           0.0                  9.00   

      pH_standard_units  Secchi_Depth_m  Water_Depth_m  Water_Temp_?C  \
349                 7.6            0.65           0.95           18.0   
365                 7.7            0.55           0.55           43.0   
371                 7.4            1.75           1.85           34.5   
372                 7.6            0.50           1.65           18.0   
373                 7.3            0.55           1.55           17.0   
...                 ...             ...            ...            ...   
2362                6.5            0.60           1.30           16.0   
2363                6.5            1.10           1.20           13.0   
2364                6.5            0.90           1.30           20.0   
2365                6.5            1.20           1.40           11.0   
2367                7.0            0.30           0.60           18.0   

      Air_Temp-Celsius  Air_Temp_?F Time_24:00    Field_Tech DateVerified  \
349                NaN          NaN       0:00  Not Recorded          NaN   
365                NaN         72.0       0:00  Not Recorded          NaN   
371                NaN          NaN      12:00  Not Recorded          NaN   
372                NaN          NaN       0:00  Not Recorded          NaN   
373                NaN          NaN       0:00  Not Recorded          NaN   
...                ...          ...        ...           ...          ...   
2362               NaN         58.0      10:15       Sue Poe   11/12/2019   
2363               NaN         49.0      10:15       Sue Poe   11/12/2019   
2364               NaN         65.0      10:22       Sue Poe   11/12/2019   
2365               NaN         42.0      10:15       Sue Poe   11/12/2019   
2367               NaN         58.0      09:30       Sue Poe   11/13/2019   

          WhoVerified  AirTemp_C  Year  
349               NaN -17.777778  1999  
365               NaN  22.222222  1999  
371               NaN -17.777778  1999  
372               NaN -17.777778  1999  
373               NaN -17.777778  1999  
...               ...        ...   ...  
2362  Christine Folks  14.444444  2018  
2363  Christine Folks   9.444444  2018  
2364  Christine Folks  18.333333  2018  
2365  Christine Folks   5.555556  2018  
2367  Christine Folks  14.444444  2018  

[1391 rows x 17 columns]

Many locations has Salinity ppt with 0 value Site id's are A,C,D & Bay

## 11. Dig a little deeper into #2, and write why there may be some uncertainty in your answer? (hint: certainty is improved by consistency in data)

In [80]:
df1.query('Salinity_ppt == 0 & Dissolved_Oxygen_mgL == 0')

Site_Id Unit_Id  Read_Date  Salinity_ppt  Dissolved_Oxygen_mgL  \
1937       A   01CSV  6/10/2014           0.0                   0.0   
1938       B   01CSV  6/10/2014           0.0                   0.0   

      pH_standard_units  Secchi_Depth_m  Water_Depth_m  Water_Temp_?C  \
1937                7.0             0.5            0.7           27.0   
1938                8.0             0.3            0.3           27.0   

      Air_Temp-Celsius  Air_Temp_?F Time_24:00 Field_Tech DateVerified  \
1937              23.0         74.0      11:00    Sue Poe    6/23/2014   
1938              74.0         23.0      10:50    Sue Poe    6/23/2014   

         WhoVerified  AirTemp_C  Year  
1937  Trenton Miller  23.333333  2014  
1938  Trenton Miller  -5.000000  2014

Solubility of dissolved oxygen also decreases as salinity increases, based on this the dissolved oxygen with minimum is 0 and salinity 0 has Site ID with A & B

## 12. Use the data to determine the correlation between Salinity (ppt) and pH (standard units). Use the DataFrame.corr(). You just need to report the correlation value.

In [81]:
df1.corr()

Salinity_ppt  Dissolved_Oxygen_mgL  pH_standard_units  \
Salinity_ppt              1.000000             -0.096862           0.271262   
Dissolved_Oxygen_mgL     -0.096862              1.000000          -0.069449   
pH_standard_units         0.271262             -0.069449           1.000000   
Secchi_Depth_m           -0.077506              0.096643          -0.130557   
Water_Depth_m            -0.119635              0.054607          -0.182553   
Water_Temp_?C             0.008669             -0.564673           0.174527   
Air_Temp-Celsius               NaN             -0.512446           0.505689   
Air_Temp_?F               0.066299             -0.560187           0.191483   
AirTemp_C                 0.039016             -0.411543           0.116325   
Year                     -0.416750              0.108474          -0.360782   

                      Secchi_Depth_m  Water_Depth_m  Water_Temp_?C  \
Salinity_ppt               -0.077506      -0.119635       0.008669   
Dissolved_Oxygen_mgL        0.096643       0.054607      -0.564673   
pH_standard_units          -0.130557      -0.182553       0.174527   
Secchi_Depth_m              1.000000       0.810988      -0.070021   
Water_Depth_m               0.810988       1.000000      -0.008880   
Water_Temp_?C              -0.070021      -0.008880       1.000000   
Air_Temp-Celsius           -0.144087      -0.076694       0.734458   
Air_Temp_?F                -0.079466      -0.019103       0.860244   
AirTemp_C                  -0.084628      -0.027647       0.692660   
Year                        0.013742       0.038773      -0.058214   

                      Air_Temp-Celsius  Air_Temp_?F  AirTemp_C      Year  
Salinity_ppt                       NaN     0.066299   0.039016 -0.416750  
Dissolved_Oxygen_mgL         -0.512446    -0.560187  -0.411543  0.108474  
pH_standard_units             0.505689     0.191483   0.116325 -0.360782  
Secchi_Depth_m               -0.144087    -0.079466  -0.084628  0.013742  
Water_Depth_m                -0.076694    -0.019103  -0.027647  0.038773  
Water_Temp_?C                 0.734458     0.860244   0.692660 -0.058214  
Air_Temp-Celsius              1.000000    -0.204001  -0.204001 -0.646594  
Air_Temp_?F                  -0.204001     1.000000   1.000000 -0.065935  
AirTemp_C                    -0.204001     1.000000   1.000000  0.030111  
Year                         -0.646594    -0.065935   0.030111  1.000000

Correlation between Salinity & pH is 0.271262